.github/workflows/rust.yml

```diff
@@ -35,12 +35,6 @@ jobs:
       - name: Remove pre-generated prost files to force regeneration
         run: rm proto/*.rs
 
-      - name: Specify dependency version for old Rust
```

Comment:
> Could you please explain a bit why we use different Cargo.toml in CI and codebase? Thanks! It is a little bit confusing to have different Cargo.toml in the same project.

For the MSRV actually. It proved `pprof-rs` can actually work on 1.56 with some dependency specified, so this only modified the dependencies for CI with 1.56 rust.

I'm not sure whether the dependencies of `pprof-rs` will be automatically updated when running `cargo update -p pprof`. If not, this behavior enables the user to update `pprof-rs` even when they are on 1.56 rust. (Even if not, it's also possible for them to downgrade the dependencies of `pprof-rs` to solve this problem)

Reply:
> Got it. Then probably the better solution is we set the correct version in the `Cargo.toml` instead of letting the user manually downgrade the deps of `pprof-rs`.

I also don't want to specify a too low version (of dependencies) for users, because it will avoid the users from using newer dependencies. It cloud also cause the dependency conflict, so I usually use `x.y` but not `x.y.z` in most cases. Do you have any better idea on this? (Upgrading MSRV can be a choice, but it's actually avoidable, by specifying dependencies manually :facepalm:  .)

> And `Cargo` does not consider the `rust-version` of `Cargo.toml` when it resolves the deps.

This is exactly the problem. The `cargo` will not automatically choose the newest **valid** version. As a developer of library, I also cannot specify different dependencies for different rust version.

> Because there are some deps of **pprof-rs** that require a higher rust toolchain to build it.

All of them are here: inferno and criterion. And the solution is to downgrade them (by the user manually, as shown in the `github.yml`). The lower version are allowed in the range specified in `Cargo.toml` of `pprof-rs`. The `sed` in ci only sets the version precisely (from `x.y` to `x.y.z`).

Reply:
> And there are no tips for users to work around https://github.com/tikv/pprof-rs/pull/204#discussion_r1142838344

Yes :facepalm: . We should add document for it.